File: 509-fibonacci-number/509-fibonacci-number.java
class Solution {
    public int fib(int n) {
        if(n == 0 || n == 1) {
            return n;
        }
        
        int prev1 = 1, prev2 = 0;
        for(int i = 2; i <= n; i++) {
            int curr = prev1 + prev2;
            prev2 = prev1;
            prev1 = curr;
        }
        
        return prev1;
    }
}

File: 509-fibonacci-number/509-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if(n < 2)
            return n;
        
        int a = 0, b = 1, c;
        while(n-- > 1) {
            c = a + b;
            a = b;
            b = c;
        }
        
        return c;
    }
};